src/video/emscripten/SDL_emscriptenvideo.c

```diff
@@ -293,12 +301,30 @@ static bool Emscripten_CreateWindow(SDL_VideoDevice *_this, SDL_Window *window,
     }
 
     selector = SDL_GetHint(SDL_HINT_EMSCRIPTEN_CANVAS_SELECTOR);
-    if (!selector) {
-        selector = "#canvas";
+    if(!selector) {
```

Comment:
```suggestion
    if (!selector || !*selector) {
```
In general, hints set to "" are equivalent to being unset. This is necessary for clearing environment overrides on some platforms.